### PR TITLE
std::tuple support (Resolving #103)

### PR DESCRIPTION
**Pull request by beojan**
This adds a `get_` lazy function that's essentially an adapted `std::get`.
Instead of providing an `unpack` lazy function, I defined `boost::phoenix::placeholders::uarg1` through `uarg10` (actually `BOOST_PHOENIX_ARG_LIMIT`) that unpack `arg1`.

### CODE REVIEW

File: test/tuple/tuple.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        BOOST_TEST(get_<double>(arg1)(tup) == std::get<double>(tup));
+        BOOST_TEST(get_<char>(arg1)(tup) == std::get<char>(tup));
+        return;
+    }
+}
+
+int main() {
+    tuple_get_test();
+    boost::report_errors();
+    return 0;
```
Comment (Kojoley) [inline test/tuple/tuple.cpp:35]: The right usage is `return boost::report_errors();` otherwise the test will silently pass.

Comment (beojan) [reply test/tuple/tuple.cpp]: Oops. Fixed.

**Comment by beojan** [comment]
The remaining AppVeyor failures appear to be a configuration issue (it can't find the compiler).

**Comment by beojan** [comment]
The remaining Travis CI failures also seem to be configuration issues for the macOS CI.

**Comment by djowel** [comment]
Odd failures!

**Comment by beojan** [comment]
For Travis: I think you need to update the macOS version used to Mojave or later for Homebrew support
For Appveyor: #105 is to fix this, I think

File: include/boost/phoenix/stl/tuple.hpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+#include <boost/phoenix/core/call.hpp>
+#include <boost/phoenix/core/expression.hpp>
+#include <boost/phoenix/core/limits.hpp>
+#include <boost/phoenix/core/meta_grammar.hpp>
+
+// Lazy functions for std::tuple (and similar)
+// get will work wherever it is accessible through ADL
+// Cribbing from functions in object
+
+namespace boost {
```
Comment (djowel) [inline include/boost/phoenix/stl/tuple.hpp:27]: Please copy the coding style of the codebase. See the other header files.

Comment (beojan) [reply include/boost/phoenix/stl/tuple.hpp]: I think I've addressed this. Let me know if there are other changes required.

**Comment by djowel** [comment]
Merged. Thanks for your contribution!

File: include/boost/phoenix/stl/tuple.hpp [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    {
+        return expression::get_with_idx<tuple_detail::idx_wrap<N>, Tuple>::make(
+        tuple_detail::idx_wrap<N>(), t);
+    }
+
+    // Make unpacked argument placeholders
+    namespace placeholders {
+        #define BOOST_PP_LOCAL_LIMITS (1, BOOST_PHOENIX_ARG_LIMIT)
+        #define BOOST_PP_LOCAL_MACRO(N)                                                \
+            auto uarg##N =                                                             \
```
Comment (jcelerier) [inline include/boost/phoenix/stl/tuple.hpp:113]: This causes a multiple definition error as soon as multiple files include this

Comment (beojan) [reply include/boost/phoenix/stl/tuple.hpp:113]: See #112 which fixes this.

**Pull request merged**